packages/palette/src/elements/Carousel/Carousel.tsx

```diff
@@ -0,0 +1,174 @@
+  color: ${color("black30")};
+  transition: opacity 250ms, color 250ms;
+
+  > svg {
+    fill: currentColor;
+  }
+
+  &:hover,
+  &:focus {
+    outline: 0;
+    color: ${color("black100")};
+  }
+
+  &:disabled {
+    opacity: 0;
+    pointer-events: none;
+  }
+`
+
+const Next = styled(Arrow)`
+  right: 0;
+  transform: translateX(100%);
+`
+
+const Previous = styled(Arrow)`
+  left: 0;
+  transform: translateX(-100%);
+`
+
+const Cell = styled(Box)``
+
+export interface CarouselProps extends BoxProps {
+  children: JSX.Element | JSX.Element[]
+}
+
+/** Carousel */
+export const Carousel: React.FC<CarouselProps> = ({ children, ...rest }) => {
+  const viewportRef = useRef<HTMLDivElement | null>(null)
+  const startRef = useRef<HTMLButtonElement | null>(null)
+  const endRef = useRef<HTMLButtonElement | null>(null)
+
+  const cells = useMemo(
+    () =>
+      Children.toArray(children)
+        .filter(isValidElement)
+        .map(child => ({ child, ref: createRef<HTMLDivElement>() })),
+    []
+  )
+
+  const [pages, setPages] = useState([0])
+
+  const { index, handleNext, handlePrev, setCursor } = useCursor({
+    max: pages.length,
+  })
+
+  const init = () => {
+    const { current: viewport } = viewportRef
+    const els = cells.map(({ ref }) => ref.current)
+    const values = els.map(node => node.clientWidth)
+    setCursor(0)
+    setPages(paginate(viewport.clientWidth, values))
+  }
+
+  useEffect(() => {
+    init()
+    window.addEventListener("resize", init)
+    return () => {
+      window.removeEventListener("resize", init)
+    }
+  }, [])
+
+  const skipToEnd = () => {
+    if (!endRef.current) return
+    endRef.current.focus()
+    setCursor(pages.length - 1)
+  }
+
+  const skipToStart = () => {
+    if (!startRef.current) return
+    startRef.current.focus()
+    setCursor(0)
+  }
+
+  return (
+    <Container {...rest}>
+      <Skip ref={startRef} onClick={skipToEnd} mb={1}>
+        Skip to end of content
+      </Skip>
+
+      <nav>
+        <Previous onClick={handlePrev} disabled={index === 0}>
+          <ChevronIcon direction="left" width={30} height={30} />
+        </Previous>
+
+        <Next onClick={handleNext} disabled={index === pages.length - 1}>
+          <ChevronIcon direction="right" width={30} height={30} />
+        </Next>
+      </nav>
+
+      <Viewport ref={viewportRef as any}>
+        <Rail as="ul" style={{ transform: `translateX(-${pages[index]}px)` }}>
+          {cells.map(({ child, ref }, i) => (
+            <Cell
+              as="li"
+              key={i}
+              ref={ref as any}
+              pr={i !== cells.length - 1 && 2}
+            >
+              {child}
+            </Cell>
+          ))}
+        </Rail>
+      </Viewport>
+
+      <Skip ref={endRef} onClick={skipToStart} mt={1}>
+        Skip to beginning of content
+      </Skip>
+
+      <VisuallyHidden aria-live="polite" aria-atomic="true">
+        Page {index + 1} of {pages.length}
+      </VisuallyHidden>
```

Comment:
This comes from the W3C guidelines: as you navigate, the aria-live content gets announced. Here, it just tells you what page you're on. The `polite` setting means that it waits for everything else to finish announcing before announcing the changed live content.